### ISSUE TITLE
php.composer should ensure that curl is installed

### ISSUE THREAD

**Issue by babilen** [saltstack-formulas/php-formula#39]
The `php.composer` state requires curl to be installed, but does not ensure it in any way.

**Comment by gravyboat** [comment]
Hmm, should the composer state handle that? Or should it simply include a state coming from another formula?

**Comment by babilen** [comment]
I am not sure and would, personally, prefer a dependency on a different formula in general. This "curl-formula" would be most trivial though and it would probably make sense to simply include a "php_curl_pkg" pkg.installed state. Either way I wouldn't mind implementing this, but didn't get around to do it when I ran into this issue.

What do you think @gravyboat ?

**Comment by ross-p** [comment]
@babilen good catch.

I agree with you that installing curl is minimal and probably better done with something like `php_curl_pkg` rather than adding an external dependency to a simple curl-formula.

**Comment by gravyboat** [comment]
@babilen Yeah as much as I hate to say it, I'd prefer if we just installed the dependency for the formula in the formula. I'm imagining some new user going 'okay I need the curl formula, oh now I need the os install formula' and so on and so forth. Whiteinge was kind enough to already create https://github.com/saltstack-formulas/php-formula/blob/master/php/curl.sls so I say just update the include in the php.composer state.
